include/deal.II/grid/grid_tools.h

```diff
@@ -2142,6 +2162,453 @@ namespace GridTools
             }
         }
   }
+
+
+
+  namespace internal
+  {
+    namespace ProjectToObject
+    {
+      struct CrossDerivative
+      {
+        const unsigned int direction_0;
+        const unsigned int direction_1;
+
+        CrossDerivative(const unsigned int d0, const unsigned int d1);
+      };
+
+      inline
+      CrossDerivative::CrossDerivative(const unsigned int d0, const unsigned int d1)
+        :
+        direction_0 (d0),
+        direction_1 (d1)
+      {}
+
+
+
+      template <typename F>
+      inline
+      auto
+      centered_first_difference(const double  center,
+                                const double  step,
+                                const F      &f)
+      -> decltype(f(center) - f(center))
+      {
+        return (f(center + step) - f(center - step))/(2.0*step);
+      }
+
+
+
+      template <typename F>
+      inline
+      auto
+      centered_second_difference(const double  center,
+                                 const double  step,
+                                 const F      &f)
+      -> decltype(f(center) - f(center))
+      {
+        return (f(center + step) - 2.0*f(center) + f(center - step))/(step*step);
+      }
+
+
+
+      template <int structdim, typename F>
+      inline
+      auto
+      cross_stencil
+      (const CrossDerivative                                        cross_derivative,
+       const Tensor<1, GeometryInfo<structdim>::vertices_per_cell> &center,
```

Comment:
Does this actually work? We may not be instantiating `Tensor<1,8>`, for example.

Reply:
This is only ever called below for `structdim = 1` and `structdim = 2`. Since it is an internal function I think that we are okay.